### PR TITLE
Improved image thumbnails and other minor fixes

### PR DESCRIPTION
**Pull request by o-smirnov**
Generate thumbnails if possible, else falls back on  tag.

Added port choice to run-padre.sh script. When making a release, please check the script, it may need to be modified (i.e. don't need to add padre to PYTHONPATH in the release version, surely).

### CODE REVIEW

File: run-padre.sh [hunk 1]
```diff
@@ -3,14 +3,25 @@
...
+  echo "is the case."
+  echo
+  echo "Since you're logged in via ssh, so I'm not opening a web browser for you. Please"
+  echo "manually browse to localhost:$port. You will probably want to employ ssh"
+  echo "port forwarding if you want to browse this notebook from your own machine, "
+  echo "e.g. log in with"
+  echo "          $ ssh -L $port:localhost$port user@machine"
+  echo 
 else
   opts = ""
```
Comment (gijzelaerr) [inline run-padre.sh]: this should be `opts=""`. I'll fix it

File: run-padre.sh [hunk 2]
```diff
@@ -3,14 +3,25 @@
 # add the directory where run-padre.sh resides to PYTHONPATH
 export PYTHONPATH=$PYTHONPATH:${0%/*}
```
Comment (gijzelaerr) [inline run-padre.sh]: does this work for you? for me `${0%/*}` becomes `.`. Which doesn't seem to work with the ipython notebook. Was there something wrong with the `pwd` statement?

Comment (o-smirnov) [inline run-padre.sh]: ${0%/*} is meant to strip /run-padre.sh from the path of the script. So if
you run /path/to/padre/run-padre.sh from any other dir, /path/to/padre gets
added to PYTHONPATH.

Comment (gijzelaerr) [inline run-padre.sh]: This doesn't work properly if you call the script as `./run-padre.sh`

Comment (gijzelaerr) [inline run-padre.sh]: maybe better to add:

```
DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
cd ${DIR}
```

at the top now, so it works from wherever you call the script.

http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in

**Pull request merged**